### DISCUSSION TITLE
FedCM

### DISCUSSION THREAD

**Discussion by iMogs370** [abacritt/angularx-social-login#753] [General]
Will this package still work when the FedCM and third-party cookies disabling come into play?

**Comment by juanmaldonadodev** [comment]
More information about this:

https://developers.google.com/identity/gsi/web/guides/fedcm-migration#migrate_your_web_app

Updates on Auto Sign-in feature
[Federated Credential Management (FedCM) Beta](https://developers.googleblog.com/2023/08/announcing-federated-credential-management-beta-for-gis.html) for Google Identity Services was launched in August 2023. Many developers tested the API and have provided valuable feedback.

One response Google heard from developers is about the FedCM automatic sign-in flow user gesture requirement. For improved privacy, Chrome requires users to reconfirm that they want to sign in to the website with Google Account in each Chrome instance even if the user approved the website prior to the FedCM rollout. This one-time reconfirmation is achieved through a single click of the One Tap prompt to demonstrate user intent to sign-in. This change may cause an initial disruption in automatic sign-in conversion rates for some websites.

Recently in M121, Chrome made a [change](https://developers.google.com/privacy-sandbox/3pcd/fedcm-updates#chrome_121_december_2023) to the FedCM automatic sign-in flow UX. The reconfirmation is only required when third-party cookies are restricted. This means:

Before third-party cookie restrictions are ramped up to 100% in [2024 Q3](https://developers.google.com/privacy-sandbox/blog/cookie-countdown-2023oct), FedCM automatic sign-in does not require reconfirmation for returning users. If users reconfirm with FedCM UI, this reconfirmation will count toward the user gesture requirement for the post-3PCD era.

FedCM automatic sign-in will check the reconfirmation status when the third-party cookies are manually restricted by users today, or by default in future Chrome.

With this change, we recommend all automatic sign-in developers migrate to FedCM as soon as possible, to reduce disruption to automatic sign-in conversion rates.

For the automatic sign-in flow, GIS JavaScript won't trigger FedCM on an older Chrome (before M121), even if your website chooses to opt-in FedCM.

**Reply by juanmaldonadodev** [reply]
I have added a pull request based on this:
https://github.com/abacritt/angularx-social-login/pull/755/commits

**Reply by valeriyalebedev** [reply]
hello @juanico18  Could you patch the version 1.3.2 as well with this change?  
we cannot upgrade the angular version for the moment but we would like to use this lib with the third party cookies blocked fix.

**Reply by akasthuri** [reply]
Hi @juanmaldonadodev,
We also use ver 1.3.2. Can you please issue a patch for this version. We received an email from Google and they will move our application to FedCM in July 2024 if we don't explicitly disable FedCM.

**Reply by juanmaldonadodev** [reply]
Hi @valeriya13 @akasthuri 
I haven't got permissions to do that. You just have to create a pull request to the affected branch and the owner repository should review it.

**Reply by akasthuri** [reply]
thanks @juanmaldonadodev for the feedback.

@valeriya13 

Due to older Angular 15 that we have, we cannot upgrade to the latest abacritt version yet. I am trying to apply a custom fix to the tag version 2.0.0 which is the closest version to 1.3.2 that supports angular 15 (Until we upgrade to Angular 17).

Extra changes required to get the abacritt library version 2.0.0 built locally;

**************************************************************************
Package.json dev dependencies need to be updated to 
"@types/google.accounts": "^0.0.14",

Original value was "@types/google.accounts": "^0.0.6",

*******************************************************************************

Second error was caused by

projects/lib/src/directives/google-signin-button.directive.ts:47:13 - error TS2322: Type 'string' is not assignable to type 'number'.

47             width: this.width,
               ~~~~~

  node_modules/@types/google.accounts/index.d.ts:780:13
    780             width?: number;


This was addressed by using “any” instead of “string” for width attribute 
In “projects/lib/src/directives/google-signin-button.directive.ts”
  @Input()
  width: any = '';


These two mismatches are already addressed by the latest abacritt library version
*******************************************************
